kebechet/managers/manager.py

```diff
@@ -60,13 +60,17 @@ def _init_igitt(service_type: ServiceType = None, service_url: str = None) -> st
 class ManagerBase:
     """A base class for manager instances holding common and useful utilities."""
 
-    def __init__(self, slug, service_type: ServiceType = None, service_url: str = None, token: str = None):
+    def __init__(self, slug, service_type: ServiceType = None, service_url: str = None, parsed_payload: dict = None,
+                 token: str = None):
         """Initialize manager instance for talking to services."""
         self.service_type = service_type or ServiceType.GITHUB
         # This needs to be called before instantiation of SourceManagement due to changes in global variables.
         self.service_url = _init_igitt(service_type, service_url)
         # Allow token expansion from env vars.
         self.slug = slug
+        if parsed_payload:
+            self.event = parsed_payload['event']
+            self.raw_payload = parsed_payload['raw_payload']
```

Comment:
I think these correspondents to the `ParsedPayload` class. Why do you duplicate it here? The same for event.

Reply:
This allows the child manager classes access these variables right? How do want to pass these values to the child classes?

Reply:
`parsed_payload` is a parameter, you can access its attributes:

`parsed_payload.raw_payload`

Reply:
Umm I thought, this would not break the `run` method, the one that uses cron job, so I passed this as a separate parameter and the `if` condition.
So I will just save the `parsed_payload` as a variable in the base class right?

Reply:
@fridex I made the changes, does that look good?

Reply:
I don't see your point in "breaking the run method" - previously the implementation did not use webhook at all - that's why there was nothing to pass to the `run` method. Now managers will be using webhooks, so they need to know webhook payload. From design point of view - webhook is not proparty/attribute of manager, but its it parameter.

However, if you want to enforce the current implementation and its the only thing that prevents us from using gitlab, I'm fine with merging this one.